ceph/rados/core_workflows.py

```diff
@@ -1545,6 +1545,57 @@ def set_mclock_parameter(
                 return False
         return True
 
+    def update_osd_state_on_cluster(self, osd_id, state):
+        """
+        Method to update the state of the OSD in the cluster. This method can mark the osd "in, out, up, down"
+        Args:
+            osd_id: ID of the osd to be operated
+            state: Property that needs to be updated.
+                  Allowed states: "in, out, up, down"
+        Returns: True -> pass, false -> fail
+        """
+        allowed_states = {"in", "out", "up", "down"}
```

Comment:
Is there a reason, why this is not a list?

Reply:
That's the max amount of allowed states, and no more would be added or removed.

Did not see any reason to make it a mutable element.